### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by fabian202**

### CODE REVIEW

File: src/components/organisms/UserForm/index.js [hunk 1]
```diff
@@ -157,16 +186,26 @@ const UserForm = ({ roles, organizations, onSave, loading, onCancel }) => {
...
                 <>
-                  <FaPen className="mr-1" /> Create
+                  <FaPen className="mr-1" /> {user ? 'Update' : 'Create'}
                 </>
               )}
             </Button>
-            <CancelButton variant="secondary" type="button" className="m-2 px-3 py-2" onClick={onCancel}>
+            <CancelButton
+              variant="secondary"
+              type="button"
```
Comment (normandanielmq) [inline src/components/organisms/UserForm/index.js:205]: Keep consistent the cancel button style by removing these lines

File: src/components/organisms/UserForm/index.js [hunk 2]
```diff
@@ -157,16 +186,26 @@ const UserForm = ({ roles, organizations, onSave, loading, onCancel }) => {
...
+              className="m-2 px-3 py-2"
+              onClick={onCancel}
+            >
               <FaTimes className="mr-1" />
```
Comment (normandanielmq) [inline src/components/organisms/UserForm/index.js:209]: This icon also breaks the consistent style

File: src/components/organisms/UserForm/schema.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ yup.addMethod(yup.string, 'exists', function (message) {
     name: 'exists',
     exclusive: true,
     message,
-    test: async function (value) {
+    test: async function (value, ctx) {
```
Comment (normandanielmq) [inline src/components/organisms/UserForm/schema.js:20]: What is ctx? Do we have a better name?

File: src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js [hunk 4]
```diff
@@ -103,23 +119,43 @@ const UsersAdministration = () => {
     })
   }
 
+  const handleNew = () => {
```
Comment (normandanielmq) [inline src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js:122]: We don't have any advantage on defining handlers outside the render component if we don't use useCallback

**Review by cesar82 — APPROVED** [review]

File: src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js [hunk 5]
```diff
@@ -106,9 +106,9 @@ const UsersAdministration = () => {
     }
   }, [errorDelete])
 
-  const handleSave = (data) => {
+  const handleSave = useCallback((data) => {
     data.user.id ? updateUser({ data }) : createUser({ data })
-  }
+  })
```
Comment (normandanielmq) [inline src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js:111]: `useCallback` without dependencies does nothing.
We have include an array of dependencies:
```
  const handleSave = useCallback((data) => {    
    data.user.id ? updateUser({ data }) : createUser({ data })  
  }, [updateUser, createUser])

File: src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js [hunk 6]
```diff
@@ -119,17 +119,17 @@ const UsersAdministration = () => {
     })
   }
 
-  const handleNew = () => {
+  const handleNew = useCallback(() => {
     setUserToEdit(null)
     handleShow()
-  }
+  })
```
Comment (normandanielmq) [inline src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js:125]: `}, [setUserToEdit, handleShow])`

btw: handleShow also needs its own useCallback

File: src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js [hunk 7]
```diff
@@ -119,17 +119,17 @@ const UsersAdministration = () => {
...
 
-  const handleEdit = (id) => {
+  const handleEdit = useCallback((id) => {
     const user = dataUsers?.users.find((u) => u.id == id)
     setUserToEdit(user)
     setUpdateId(user.id)
     handleShow()
-  }
+  })
```
Comment (normandanielmq) [inline src/components/pages/ProgramAdministrationPage/UsersAdministration/index.js:132]: `}, [dataUsers, setUserToEdit, setUpdateId, handleShow])`

**Pull request merged**